src/DocumentFormat.OpenXml.Framework/OpenXmlPartWriter.cs

```diff
@@ -365,5 +471,179 @@ public override void Close()
         }
 
         #endregion
+
+        // Async Methods
+#if FEATURE_ASYNC_SAX_XML
+        /// <summary>
+        /// Asynchronously writes the XML declaration with the version "1.0".
+        /// </summary>
+        public async override Task WriteStartDocumentAsync()
+        {
+            ThrowIfObjectDisposed();
+            await _xmlWriter.WriteStartDocumentAsync().ConfigureAwait(true);
+            return;
+        }
```

Comment:
```suggestion
        public override Task WriteStartDocumentAsync()
        {
            ThrowIfObjectDisposed();
            return _xmlWriter.WriteStartDocumentAsync();
        }
```

For these kind of methods, we don't need to actually await, but rather just pass the task back.

The only time that doesn't work is if you need to do something with it after (including if you have any using statements)